app/Filament/Resources/AppointmentResource.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Filament\Resources;
+
+use App\Filament\Resources\AppointmentResource\Pages;
+use App\Filament\Resources\AppointmentResource\RelationManagers;
+use App\Models\Appointment;
+use Filament\Forms;
+use Filament\Forms\Components\DatePicker;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\Textarea;
+use Filament\Forms\Components\TextInput;
+use Filament\Forms\Form;
+use Filament\Resources\Resource;
+use Filament\Tables;
+use Filament\Tables\Columns\TextColumn;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\SoftDeletingScope;
+
+class AppointmentResource extends Resource
+{
+    protected static ?string $model = Appointment::class;
+
+    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';
+
+    public static function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Select::make('patient_id')
+                    ->relationship('patient', 'name')
+                    ->label('Patient')
+                    ->required()
+                    ->columnSpan('full'),
+                TextInput::make('patient_first_name')
+                    ->required(),
+                TextInput::make('patient_last_name')
+                    ->required(),
+                Select::make('department_id')
+                    ->relationship('department', 'name')
+                    ->label('Department')
+                    ->required(),
+                Select::make('doctor_id')
+                    ->relationship('doctor', 'name')
+                    ->label('Doctor')
+                    ->required(),
+                Textarea::make('notes')
+                    ->columnSpan('full'),
+                DatePicker::make('appointment_date')
+                    ->required()
+                    ->native(false)
+                    ->default(today())
+                    ->minDate(today()),
+                Select::make('status')
+                    ->options([
+                        'confirmed' => 'Confirmed',
+                        'pending' => 'Pending',
+                        'cancelled' => 'Cancelled'
+                    ])
+                    ->required(),
+            ]);
+    }
+
+    public static function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                TextColumn::make('patient.name')
+                    ->searchable()
+                    ->sortable(),
+                TextColumn::make('doctor.name')
+                    ->searchable()
+                    ->sortable(),
+                TextColumn::make('department.name')
+                    ->searchable()
+                    ->sortable(),
+                TextColumn::make('appointment_date')
+                    ->date()
+                    ->sortable(),
+                TextColumn::make('status')
```

Comment:
Give ability to update status from here as well. Because just to update status i am going to edit page.